aws/scenarios/vm/ec2VM/ec2VM.go

```diff
@@ -0,0 +1,84 @@
+package ec2vm
+
+import (
+	"github.com/DataDog/test-infra-definitions/aws"
+	awsEc2 "github.com/DataDog/test-infra-definitions/aws/ec2/ec2"
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/vm/agentinstall"
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/vm/os"
+	"github.com/DataDog/test-infra-definitions/command"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ec2"
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type Ec2VM struct {
+	runner *command.Runner
+}
+
+// NewEc2VM creates a new EC2 instance. By default use WithOS(os.UbuntuOS, os.AMD64Arch).
+func NewEc2VM(ctx *pulumi.Context, options ...func(*Params) error) (*Ec2VM, error) {
```

Comment:
What if I need two VMs? a Name parameter would appropriate and would avoid hardcoding names.

Reply:
Code updated.